src/match.c

```diff
@@ -1,128 +1,199 @@
 #include <stdio.h>
 #include <stdlib.h>
-#include <stdbool.h>
-#include <assert.h>
-#include <gmodule.h>
+#include <string.h>
+#include "simclist.c"
+#include "simclist.h"
 #include "match.h"
 
 /* See match.h for descriptions of functions */
 
-match* new_match(char* word, int line)
-{
+match* match_new(char* word, int lineNum, int position, char* line) {
     match* m = malloc(sizeof(match));
     int rc;
-    
-    if (m == NULL) 
-    {
+    if (m == NULL) {
         printf("could not allocate memory for match\n");
         return NULL;
     }
-    
-    rc = init_match(m, word, line);
-    if (rc != 0) 
-    {
-        printf("Could not init match word %s on line %d\n", word, line);
+
+    rc = match_init(m, word, lineNum, position, line);
+    if (rc != 0) {
+        printf("Could not init match word %s on line [%d] position %d\n", word, lineNum, position);
         return NULL;
     }
 
     return m;
 }
 
-int init_match(match* match, char* word, int line)
-{
-    if(match == NULL) return -1;
+int match_init(match* match, char* word, int lineNum, int position, char* line) {
+    if (match == NULL) {
+        return -1;
+    }
+
+    match->word = strdup(word);
+    match->lineNum = lineNum;
+    match->position = position;
+    match->line = strdup(line);
 
-    match->word = word;
-    match->line = line;
-    
     return 0;
 }
 
-int free_match(match* match)
-{
-    if(match == NULL) return -1;
-    free(match);
+int match_set_line(match* match, char* line) {
+    if (match == NULL) {
+        return -1;
+    }
+
+    match->line = strdup(line);
     return 0;
```

Comment:
Let's have this function return `EXIT_SUCCESS` and `EXIT_FAILURE`.

Reply:
changed